health_checks.go

```diff
@@ -0,0 +1,201 @@
+package rabbithole
+
+import "strconv"
+
+type TimeUnit string
+
+const (
+	SECONDS TimeUnit = "seconds"
+	DAYS    TimeUnit = "days"
+	MONTHS  TimeUnit = "months"
+	YEARS   TimeUnit = "years"
+)
+
+type Protocol string
+
+const (
+	AMQP091   Protocol = "amqp091"
+	AMQP10    Protocol = "amqp10"
+	MQTT      Protocol = "mqtt"
+	STOMP     Protocol = "stomp"
+	WEB_MQTT  Protocol = "web-mqtt"
+	WEB_STOMP Protocol = "web-stomp"
+)
+
+// Health represents response from healthchecks endpoint
+type Health struct {
```

Comment:
This response data structure tries to cover all health checks but they have very different context. I'd suggest that we try to come up with a very small interface for health check responses (e.g. to easily check for success/failure),
and use a separate response type for every health check. This way we would be able to provide all the relevant
response context without exposing it to those users who only want to check if the operation succeeded.

WDYT?

Reply:
Yes, good idea. For me it is a little bit unclear which fields are in which response type:

General response:
```go
type Health struct {
	Status string `json:"status"`
	Error  string `json:"error"`
	Reason string `json:"reason"`
}
```

PortListenerCheck:
```go
type HealthPortListener struct {
	Status string `json:"status"`
	Error  string `json:"error"'`
	Reason string `json:"reason"`
	Port   string
	Ports  []string `json:"ports"`
}
```

ProtocolListenerCheck:

```go
type HealthProtocolListener struct {
	Status    string   `json:"status"`
	Reason    string   `json:"reason"`
	Missing   string   `json:"missing"`
	Protocols []string `json:"protocols"`
}
```
Btw this is also a little confusing. The response returns protocols:

```
"protocols":["http","http/prometheus","clustering","amqp"]
```

which aren't the same as in the api documentation:

> Valid protocol names are: amqp091, amqp10, mqtt, stomp, web-mqtt, web-stomp.

And like `/health/checks/protocol-listener` it does not return field `error`.

Reply:
Those protocols are from the list of known protocols (accepted and used by CLI tools, for example). They should be supported by any enum types we use/introduce.

There can be minor inconsistencies in the context provided by different endpoints. We expect that most users won't really use this context beyond logging. Can we easily default to a string such as "N/A" when it is missing in an endpoint response?